blockchain/fullblocktests/generate.go

```diff
@@ -2852,3 +2852,196 @@ func GeneratePhononBlocks() (tests [][]TestInstance, err error) {
+func GenerateCosmicInflationBlocks() (tests [][]TestInstance, err error) {
+	// In order to simplify the generation code which really should never
+	// fail unless the test code itself is broken, panics are used
+	// internally.  This deferred func ensures any panics don't escape the
+	// generator by replacing the named error return with the underlying
+	// panic error.
+	defer func() {
+		if r := recover(); r != nil {
+			tests = nil
+
+			switch rt := r.(type) {
+			case string:
+				err = errors.New(rt)
+			case error:
+				err = rt
+			default:
+				err = errors.New("Unknown panic")
+			}
+		}
+	}()
+
+	// Create a test generator instance initialized with the genesis block
+	// as the tip.
+	g, err := makeTestGenerator(regressionNetParams)
+	if err != nil {
+		return nil, err
+	}
+
+	acceptBlock := func(blockName string, block *wire.MsgBlock, isMainChain, isOrphan bool) TestInstance {
+		blockHeight := g.blockHeights[blockName]
+		return AcceptedBlock{blockName, block, blockHeight, isMainChain,
+			isOrphan}
+	}
+
+	coinbaseMaturity := g.params.CoinbaseMaturity
+	var testInstances []TestInstance
+	for i := uint16(0); i < coinbaseMaturity; i++ {
+		blockName := fmt.Sprintf("bm%d", i)
+		g.nextBlock(blockName, nil)
+		g.saveTipCoinbaseOut()
+		testInstances = append(testInstances, acceptBlock(g.tipName,
+			g.tip, true, false))
+	}
+	tests = append(tests, testInstances)
+
+	// Collect spendable outputs.  This simplifies the code below.
+	var outs []*spendableOut
+	for i := uint16(0); i < coinbaseMaturity; i++ {
+		op := g.oldestCoinbaseOut()
+		outs = append(outs, &op)
+	}
+
+	add64BitInteger := func(block *wire.MsgBlock) {
+		// unsortedTxs is a slice of bchutil.Tx's that we will sort later using CTOR.
+		var unsortedTxs []*bchutil.Tx
+
+		// Let's first add any txs (excluding the coinbase) into unsortedTxs
+		for _, tx := range block.Transactions[1:] {
+			unsortedTxs = append(unsortedTxs, bchutil.NewTx(tx))
+		}
+
+		// Create one tx paying a p2sh output
+		tx1 := createSpendTx(outs[0], 0)
+		builder := txscript.NewScriptBuilder().
+			AddOp(txscript.OP_1).
+			AddOp(txscript.OP_ADD).
+			AddInt64(1152921504606846976).
+			AddOp(txscript.OP_EQUAL)
+		redeemScript, err := builder.Script()
+		if err != nil {
+			panic(err)
+		}
+
+		addr, err := bchutil.NewAddressScriptHash(redeemScript, &chaincfg.RegressionNetParams)
+		if err != nil {
+			panic(err)
+		}
+		script, err := txscript.PayToAddrScript(addr)
+		if err != nil {
+			panic(err)
+		}
+		tx1.TxOut[0].PkScript = script
+
+		block.AddTransaction(tx1)
+		unsortedTxs = append(unsortedTxs, bchutil.NewTx(tx1))
+
+		// Spend from tx1
+		so1 := &spendableOut{
+			amount: bchutil.Amount(tx1.TxOut[0].Value),
+			prevOut: wire.OutPoint{
+				Hash:  tx1.TxHash(),
+				Index: 0,
+			},
+		}
+
+		tx2 := createSpendTx(so1, 0)
+		scriptSig, err := txscript.NewScriptBuilder().
+			AddInt64(1152921504606846975).
+			AddData(redeemScript).Script()
```

Comment:
```suggestion
		tx1 := createSpendTx(outs[0], 0)
		sixtyBitInt := int64(2 ^ 60)
		builder := txscript.NewScriptBuilder().
			AddOp(txscript.OP_1).
			AddOp(txscript.OP_ADD).
			AddInt64(sixtyBitInt).
			AddOp(txscript.OP_EQUAL)
		redeemScript, err := builder.Script()
		if err != nil {
			panic(err)
		}

		addr, err := bchutil.NewAddressScriptHash(redeemScript, &chaincfg.RegressionNetParams)
		if err != nil {
			panic(err)
		}
		script, err := txscript.PayToAddrScript(addr)
		if err != nil {
			panic(err)
		}
		tx1.TxOut[0].PkScript = script

		block.AddTransaction(tx1)
		unsortedTxs = append(unsortedTxs, bchutil.NewTx(tx1))

		// Spend from tx1
		so1 := &spendableOut{
			amount: bchutil.Amount(tx1.TxOut[0].Value),
			prevOut: wire.OutPoint{
				Hash:  tx1.TxHash(),
				Index: 0,
			},
		}

		tx2 := createSpendTx(so1, 0)
		scriptSig, err := txscript.NewScriptBuilder().
			AddInt64(sixtyBitInt - 1).
			AddData(redeemScript).Script()
```

Reply:
Ah shit sorry. This suggestion just came out as a big red block. It just has 3 line changes:

```
		sixtyBitInt := int64(2 ^ 60)
...
			AddInt64(sixtyBitInt).
...
			AddInt64(sixtyBitInt - 1).
```

I suggest it because it took me a while to realize the two scripts are connected.